src/query/Index.cpp

```diff
@@ -110,6 +113,36 @@ TriangleTraits::Ptr Index::IndexImpl::getTriangleRTree(const mesh::Mesh &mesh)
   return indices.triangleRTree;
 }
 
+TetrahedronTraits::Ptr Index::IndexImpl::getTetraRTree(const mesh::Mesh &mesh)
+{
+  if (indices.tetraRTree) {
+    return indices.tetraRTree;
+  }
+
+  precice::utils::Event e("query.index.getTriangleIndexTree." + mesh.getName());
+
+  // We first generate the values for the tetra rtree.
+  // The resulting vector is a random access range, which can be passed to the
+  // constructor of the rtree for more efficient indexing.
+  std::vector<TetrahedronTraits::IndexType> elements;
+  elements.reserve(mesh.tetrahedra().size());
+  for (size_t i = 0; i < mesh.tetrahedra().size(); ++i) {
+    // Instead of using bg::return_envelope, we compute the BB of the tetra ourself
```

Comment:
The comment misses the reason why

Reply:
The thing is that I didn't fully get why :sweat_smile: 
I couldn't find any reference as to why boost's get_envelope wouldn't work.

Reply:
There is a guarantee that it works and the fact that one version works doesn't mean that all versions do.

Reply:
So I just write that we don't trust Boost?

Reply:
No, we don't want to rely on undocumented behaviour.